common/src/main/java/tachyon/util/ShellUtils.java

```diff
@@ -0,0 +1,246 @@
+      mExitCode = mProcess.waitFor();
+      if (mExitCode != 0) {
+        throw new ExitCodeException(mExitCode, errMsg.toString());
+      }
+    } catch (InterruptedException ie) {
+      throw new IOException(ie.toString());
+    } finally {
+      // close the input stream
+      try {
+        // JDK 7 tries to automatically drain the input streams for us
+        // when the process exits, but since close is not synchronized,
+        // it creates a race if we close the stream first and the same
+        // fd is recycled. the stream draining thread will attempt to
+        // drain that fd!! it may block, OOM, or cause bizarre behavior
+        // see: https://bugs.openjdk.java.net/browse/JDK-8024521
+        // issue is fixed in build 7u60
+        InputStream stdout = mProcess.getInputStream();
+        synchronized (stdout) {
+          inReader.close();
+        }
+      } catch (IOException ioe) {
+        LOG.warn("Error while closing the input stream", ioe);
+      }
+      mProcess.destroy();
+    }
+  }
+
+  /** return an array containing the command name & its parameters */
+  protected abstract String[] getExecString();
+
+  /** Parse the execution result */
+  protected abstract void parseExecResult(BufferedReader lines) throws IOException;
+
+  /**
+   * get the current sub-process executing the given command
+   *
+   * @return process executing the command
+   */
+  public Process getProcess() {
+    return mProcess;
+  }
+
+  /**
+   * get the exit code
+   *
+   * @return the exit code of the process
+   */
+  public int getExitCode() {
+    return mExitCode;
+  }
+
+  /**
+   * This is an IOException with exit code added.
+   */
+  public static class ExitCodeException extends IOException {
+
+    private static final long serialVersionUID = -6520494427049734809L;
+
+    private final int mExitCode;
+
+    public ExitCodeException(int exitCode, String message) {
+      super(message);
+      mExitCode = exitCode;
+    }
+
+    public int getExitCode() {
+      return mExitCode;
+    }
+
+    @Override
+    public String toString() {
+      final StringBuilder sb = new StringBuilder("ExitCodeException ");
+      sb.append("exitCode=").append(mExitCode).append(": ");
+      sb.append(super.getMessage());
+      return sb.toString();
+    }
+  }
+
+  public interface CommandExecutor {
+
+    void execute() throws IOException;
+
+    int getExitCode() throws IOException;
+
+    String getOutput() throws IOException;
+
+    void close();
+
+  }
+
+  /**
+   * A simple shell command executor.
+   *
+   * <code>ShellCommandExecutor</code>should be used in cases where the output of the command needs
+   * no explicit parsing and where the command, working directory and the environment remains
+   * unchanged. The output of the command is stored as-is and is expected to be small.
+   */
+  public static class ShellCommandExecutor extends ShellUtils implements CommandExecutor {
```

Comment:
You can hide this class from outside. And you don't need it to extend ShellUtils, since all methods from ShellUtils should be accessible.